src/common-ssh/guac_ssh.c

```diff
@@ -449,6 +449,7 @@ guac_common_ssh_session* guac_common_ssh_create_session(guac_client* client,
     if (current_address == NULL) {
         guac_client_abort(client, GUAC_PROTOCOL_STATUS_UPSTREAM_ERROR,
                 "Unable to connect to any addresses.");
+        freeaddrinfo(addresses);
```

Comment:
Couldn't you just do this once above this block, instead of once here and once on line 458?